contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+
+The automation of these mundane tasks could make development a more enjoyable (more on that in a bit), and faster process. All of these optimizations can make developing an application between 30-40% faster on Rails, while also making it easier to learn, as you do not need to understand HTTP verbs to make a basic Rails application (while mastery of the topic will certainly help). Furthermore, programmers experienced in Rails usually find it easier to start on existing Rails projects, since the conventions followed are usually the same, giving you the same directory structure and route naming conventions everywhere. 
+  
+### Optimized for Programmer Happiness
+
+One of the primary tenets of Ruby was to make it the "Least Surprising Language", where the language was designed to perform exactly as the developer thought would be the most "expected" function. Rails built on that, and was designed to be optimized for Programmer Happiness. This meant re-imagining many of coding's most standard conventions, as well as adding significant bloat, all for the purpose of making the code look "pretty", and inherently "nice" to write (as subjective as that sounds). 
+
+For example, in order to fetch yesterday's date in Python, one may have to write
+
+```python
+import datetime
+datetime.datetime.now() - datetime.timedelta(days = 1)
+```
+
+The same action in Rails is done using
+```ruby
+1.day.ago
+```
+
+Similarly, Rails also added a new method to access an array element. Instead of following the C convention of using square-brackets to access the second element in the array "elements" like
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, intuitive, and inherently happiness-inducing.
+
+### Large and Active Community of Developers
+
+The Ruby language has a very passionate and extensive community of developers, and there is a vast array of libraries (called "Gems"), that can be used with your Rails projects. These libraries provide a lot of functionalities, and can be used to add flexibility to Rails projects when required. Furthermore, as a result of its popularity, it is very likely that any problems you encounter with Rails will have a solution that you can find on Google or StackOverflow. If not, the Ruby community is actually very famous for being 'nice', so you're likely to get a response to a new problem very quickly too. 
```

Comment:
`add flexibility to Rails projects when required.` (split these into 2 separate paragraphs) `Furthermore, as a result of its popularity`

Reply:
`If not, the Ruby community is actually very famous for being 'nice', so you're likely to get a response to a new problem very quickly too.` -> `For any new problems, you are also likely to receive a response very quickly because the Ruby community is known for being welcoming and helpful.`